transactions/WFLY-14869-microprofile-lra.adoc

```diff
@@ -0,0 +1,123 @@
+
+The MicroProfile LRA is a transaction processing pattern well suited for microservice development based on the saga
+pattern. The motivation of such processing is summarized in the
+https://github.com/eclipse/microprofile-lra/blob/1.0/README.adoc#motivation[MicroProfile Spec readme].
+
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14869
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1496
+
+=== Dev Contacts
+
+* mailto:mstefank@redhat.com[{author}]
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+
+=== Testing By
+
+* [x] Engineering
+* [ ] QE
+
+
+=== Affected Projects or Components
+
+* Transactions
+* WildFly
+
+=== Other Interested Projects
+
+* MicroProfile
+* Galleon
+* WildFly Operator
+
+=== Relevant Installation Types
+
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+* [x] Managed domain
+* [ ] OpenShift s2i
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+The feature will integrate the LRA specification into the WildFly feature pack as two new extensions (one for LRA coordinator and one for LRA participant) together with two new subsystems and Galleon layers accordingly.
+The Galleon layers will be available for provisioning.
+The provisioned layers will provide all functionality defined in LRA specification for users in WildFly.
+
+The integrations will consist of two separate subsystems - microprofile-lra-coordinator and microprofile-lra-participant) and two separate extensions - org.wildfly.extension.microprofile.lra-coordinator and org.wildfly.extension.microprofile.lra-participant.
+
+The first layer provides a capability of the LRA participant - bundling the Narayana client library capable
+to parse the annotations from LRA specification and work with JAX-RS and non-JAX-RS bindings.
+
+The second layer provides the capability of the LRA coordinator - a server which provides HTTP endpoints where
+applications may register as participants which the coordinator then calls back (via HTTP calls) when LRA finishes.
+
+The feature pack implementation will be placed amongst other MicroProfile WildFly subsystems under
+https://github.com/wildfly/wildfly/tree/main/microprofile[the WildFly GitHub repository].
+
+Both subsystems are optional to allow for exclusion during provisioning.
+
+The LRA layers will be dependent on the `transactions` subsystem. The LRA uses the same object store types
+as the JTA/JTS transaction uses, it requires the recovery manager thread to be running and requires transaction
+specifics like unique node-identifier per running application server (WildFly).
+
+As said before the LRA coordinator is based on the Narayana core (`transactions` subsystem within WildFly). The LRA records and JTA records share the same storage and recovery processing.
+
+=== Nice-to-Have Requirements
+
+Create a quickstart on LRA on how to use LRA with WildFly for https://github.com/wildfly/quickstart.
```

Comment:
Has this quickstart already been created? Do you have the Jira link to share @xstefank ?

Reply:
no, still on my TODO list

Reply:
No problem, please consider opening the Jira issue, I would be glad to help you with that quickstart!